### ISSUE TITLE
I would like to be able to translate messages (language file)

### ISSUE THREAD

**Issue by ziceptor** [MinecraftForge/MinecraftForge#4819]
main\java\net\minecraftforge\fml\common\FMLCommonHandler.java
main\java\net\minecraftforge\fml\common\network\internal\FMLNetworkHandler.java

**Comment by kierajreed** [comment]
Please add more details - file names won't do much.

**Comment by ziceptor** [comment]
https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java#L629

https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/src/main/java/net/minecraftforge/fml/common/FMLCommonHandler.java#L639

To be able to translate messages from the server into another language.

**Comment by covers1624** [comment]
Those cannot be translated, the client is what handles translations and forge needs to support vanilla clients for these messages specifically.

**Comment by ziceptor** [comment]
If you translate rows and rebuild forge, the messages in the client appear translated. Why can't I add server message localization support?

**Comment by covers1624** [comment]
The server has no way of knowing what language to translate it to, as the client selects the language not the server. And again, these messages are used for vanilla and modded connections, any aditions made by forge, such as support for knowing the clients localization would not be compatiable with vanilla so there is really no point.

**Comment by mezz** [comment]
Unfortunately covers is right, Mojang needs to add better localization support for server messages before we can localize these nicely.

**Comment by ziceptor** [comment]
Thank You!
